### PR TITLE
Update Dial Concurrency factor

### PR DESCRIPTION
**Pull request by maschad**
# Description

Increased the default dial concurrency factor to 8 based on the [bottom dialer number of connections](https://kademlia-exporter.max-inden.de/d/Pfr0Fj6Mk/rust-libp2p?orgId=1&refresh=30s) as well as the [go-libp2p uses a dial concurrency factor of 8](https://github.com/libp2p/go-libp2p/blob/master/p2p/net/swarm/swarm_dial.go?rgh-link-date=2022-06-28T03%3A34%3A24Z#L68)

## Links to any relevant issues

Part of https://github.com/libp2p/rust-libp2p/issues/2678

## Change checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: swarm/src/connection/pool.rs [hunk 1]
```diff
@@ -1175,8 +1175,8 @@ impl Default for PoolConfig {
             executor: None,
             task_event_buffer_size: 32,
             task_command_buffer_size: 7,
-            // By default, addresses of a single connection attempt are dialed in sequence.
-            dial_concurrency_factor: NonZeroU8::new(1).expect("1 > 0"),
+            // By default, addresses of a single connection attempt are dialed in sequence. Set to a default of 8 based on regular frequence of dialer connections
```
Comment (mxinden) [inline swarm/src/connection/pool.rs]:
```suggestion
```

I think now that this is no longer sequential, the comment is not needed.

File: swarm/CHANGELOG.md [hunk 2]
```diff
@@ -4,8 +4,11 @@
 
 - Extend log message when exceeding inbound negotiating streams with peer ID and limit. See [PR 2716].
 
+- Update dial address concurrency factor to `8` via `Swarm::dial_concurrency_factor`. See [PR 2741].
```
Comment (mxinden) [inline swarm/CHANGELOG.md]:
```suggestion
- Update dial address concurrency factor to `8`, thus dialing up to 8 addresses concurrently for a single connection attempt. See `Swarm::dial_concurrency_factor` and [PR 2741].
```

**Review by mxinden — COMMENTED** [review]
Two small comments, otherwise this looks good to me. Thanks!

**Comment by maschad** [comment]
Thanks for the review @mxinden , I've made the suggested changes.

**Review by mxinden — APPROVED** [review]

**Pull request merged**